beaver/transports/redis_transport.py

```diff
@@ -75,13 +76,23 @@ def callback(self, filename, lines, **kwargs):
             namespace = self._namespace
         self._logger.debug('Got namespace: ' + namespace)
 
+        data_type = self._data_type
+        self._logger.debug('Got data type: ' + data_type)
+
         server = self._get_next_server()
         self._logger.debug('Got redis server: ' + server['url'])
 
         pipeline = server['redis'].pipeline(transaction=False)
 
+        if data_type == 'list':
+            data_type_method = pipeline.rpush
+        elif data_type == 'channel':
+            data_type_method = pipeline.publish
+        else:
+            raise TransportException('Unknown Redis data type')
```

Comment:
We should do this checking in the `__init__`, though the method setting can stay here.

Something like:

```
callback_map = {
    'list': pipeline.rpush,
    'channel': pipeline.publish,
}

callback_method = callback_map[data_type]
```

Would be a bit nicer as well.